### PR TITLE
[KOGITO-6835] Mapping trigger to channel

### PR DESCRIPTION
**Pull request by fjtirado**
See JIRA description and comments
Updated example using functionality https://github.com/kiegroup/kogito-examples/pull/1152

### CODE REVIEW

**Comment by kie-ci1** [comment]
**(tests) - kogito-examples job** `#1296` was: **UNSTABLE**
Possible explanation: This should be test failures


**Test results:**
- PASSED: 452
- FAILED: 2

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.downstream.kogito-examples/1296/testReport/org.kie.kogito.quarkus.outbox/OutboxIT/___/">org.kie.kogito.quarkus.outbox.OutboxIT.(?)</a></summary>
java.lang.RuntimeException: org.testcontainers.containers.ContainerLaunchException: Timed out waiting for container port to open (localhost ports: [8080, 49183] should be listening)<br/>	at org.kie.kogito.quarkus.outbox.OutboxIT.before(OutboxIT.java:83)<br/>Caused by: org.testcontainers.containers.ContainerLaunchException: Timed out waiting for container port to open (localhost ports: [8080, 49183] should be listening)<br/>
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.downstream.kogito-examples/1296/testReport/org.kie.kogito.springboot.outbox/OutboxIT/___/">org.kie.kogito.springboot.outbox.OutboxIT.(?)</a></summary>
java.lang.RuntimeException: org.testcontainers.containers.ContainerLaunchException: Timed out waiting for container port to open (localhost ports: [8080, 49237] should be listening)<br/>	at org.kie.kogito.springboot.outbox.OutboxIT.before(OutboxIT.java:83)<br/>Caused by: org.testcontainers.containers.ContainerLaunchException: Timed out waiting for container port to open (localhost ports: [8080, 49237] should be listening)<br/>
</details>

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.downstream.kogito-examples/1296/display/redirect

**Comment by kie-ci1** [comment]
**(tests) - kogito-apps job** `#1202` was: **UNSTABLE**
Possible explanation: This should be test failures


**Test results:**
- PASSED: 2678
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.downstream.kogito-apps/1202/testReport/org.kie.kogito.index/ProcessDataIndexMongoDBIT/testProcessInstanceEvents/">org.kie.kogito.index.ProcessDataIndexMongoDBIT.testProcessInstanceEvents</a></summary>
java.lang.AssertionError: <br/>1 expectation failed.<br/>JSON path data.UserTaskInstances[0].comments.size() doesn't match.<br/>Expected: is <1><br/>  Actual: 0<br/><br/>
</details>

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.downstream.kogito-apps/1202/display/redirect

**Comment by kie-ci1** [comment]
**(tests) - kogito-examples job** `#1302` was: **UNSTABLE**
Possible explanation: This should be test failures


**Test results:**
- PASSED: 453
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.downstream.kogito-examples/1302/testReport/org.kie.kogito.quarkus.outbox/OutboxIT/testSendProcessEvents/">org.kie.kogito.quarkus.outbox.OutboxIT.testSendProcessEvents</a></summary>
org.awaitility.core.ConditionTimeoutException: <br/>Assertion condition defined as a lambda expression in org.kie.kogito.quarkus.outbox.OutboxIT 1 expectation failed.<br/>Expected status code <200> but was <404>.<br/> within 1 minutes.<br/>	at org.kie.kogito.quarkus.outbox.OutboxIT.testSendProcessEvents(OutboxIT.java:129)<br/>Caused by: java.lang.AssertionError: <br/>1 expectation failed.<br/>Expected status code <200> but was <404>.<br/><br/>	at org.kie.kogito.quarkus.outbox.OutboxIT.lambda$testSendProcessEvents$1(OutboxIT.java:136)<br/>
</details>

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.downstream.kogito-examples/1302/display/redirect

**Comment by fjtirado** [comment]
jenkins rerun kogito-examples tests

**Comment by fjtirado** [comment]
@cristianonicolai please take a look

File: quarkus/addons/messaging/deployment/src/main/java/org/kie/kogito/addon/cloudevents/quarkus/deployment/ChannelMappingStrategy.java [hunk 1]
```diff
@@ -30,31 +34,53 @@ private ChannelMappingStrategy() {
...
+        Collection<String> result = new HashSet<>();
+        result.add(channelName);
+        return result;
+    }
+
+    private static ChannelInfo getChannelInfo(String property, String prefix, boolean isInput, String defaultChannelName, Map<String, Collection<String>> triggers) {
+        String name = property.substring(prefix.length(), property.lastIndexOf('.'));
+        return new ChannelInfo(name, triggers.getOrDefault(name, Collections.singleton(name)),
+                getClassName(config.getOptionalValue(getPropertyName(prefix, name, "value." + (isInput ? "deserializer" : "serializer")), String.class)), isInput,
+                name.equals(defaultChannelName));
```
Comment (ricardozanini) [inline quarkus/addons/messaging/deployment/src/main/java/org/kie/kogito/addon/cloudevents/quarkus/deployment/ChannelMappingStrategy.java:83]: nit; Can we have auxiliary methods to resolve these property keys?

**Review by hbelmiro — APPROVED** [review]

**Pull request merged**